#ifndef MAINWINDOW_H
#define MAINWINDOW_H

#include <QMainWindow>

QT_BEGIN_NAMESPACE
namespace Ui {
class MainWindow;
}
QT_END_NAMESPACE

class MainWindow : public QMainWindow
{
    Q_OBJECT

public:
    MainWindow(QWidget *parent = nullptr);
    ~MainWindow();
    void showMessage(const QString &message, unsigned int id);
protected:
     void closeEvent(QCloseEvent *event) override;
private slots:
    void onSubmit();
    void onAdvanced();
    void onResults();
    void onDivergenceCheck();

private:
    bool hasSubdirectories();
    void loadPersisted();
    void savePersisted() const;
    Ui::MainWindow *ui;
};
#endif // MAINWINDOW_H
